spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -130,9 +131,9 @@ S3ObjectConverter s3ObjectConverter(Optional<ObjectMapper> objectMapper) {
 
 	@Bean
 	@ConditionalOnMissingBean
-	S3OutputStreamProvider diskBufferingS3StreamProvider(S3Client s3Client,
+	S3OutputStreamProvider inMemoryBufferingS3StreamProvider(S3Client s3Client,
 			Optional<S3ObjectContentTypeResolver> contentTypeResolver) {
-		return new DiskBufferingS3OutputStreamProvider(s3Client,
+		return new InMemoryBufferingS3OutputStreamProvider(s3Client,
```

Comment:
We should provide a way that `bufferSize` can be set in properties.

Reply:
it can be easily defined by overwriting a bean. i am not sure how to structure it at this stage, we can add a property later when feedback like this comes.